chia/protocols/farmer_protocol.py

```diff
@@ -27,6 +50,7 @@ class NewSignagePoint(Streamable):
     sub_slot_iters: uint64
     signage_point_index: uint8
     peak_height: uint32
+    sp_source_data: SignagePointSourceData
```

Comment:
since these changes to the harvester protocol are not a backwards compatible, I think you should:

1. separate the "protocol version" constant we have right now, into one for each protocol (full node, farmer and harvester)
2. bump the harvester protocol version
3. make sure the handshake fails and prints a clear error if the harvester is handshaking with the wrong version

Reply:
Or use new messages ids? I think that's what has been done before.

Reply:
I had asked Earle about this before I made the changes. He mentioned we should just modify existing messages and bump the version number and that we ought to just let people know they have to upgrade.

Are you sure we want to include the scope of separating the protocol versions per node type as part of this? Not necessarily opposed, but...